test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetAccountNftInfos.java

```diff
@@ -0,0 +1,136 @@
+package com.hedera.services.bdd.spec.queries.token;
+
+/*-
+ * ‌
+ * Hedera Services Test Clients
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.bdd.spec.HapiApiSpec;
+import com.hedera.services.bdd.spec.queries.HapiQueryOp;
+import com.hedera.services.bdd.spec.transactions.TxnUtils;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.NftID;
+import com.hederahashgraph.api.proto.java.Query;
+import com.hederahashgraph.api.proto.java.Response;
+import com.hederahashgraph.api.proto.java.TokenGetAccountNftInfosQuery;
+import com.hederahashgraph.api.proto.java.TokenNftInfo;
+import com.hederahashgraph.api.proto.java.Transaction;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.util.Strings;
+import org.junit.Assert;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static com.hedera.services.bdd.spec.queries.QueryUtils.answerCostHeader;
+import static com.hedera.services.bdd.spec.queries.QueryUtils.answerHeader;
+
+public class HapiGetAccountNftInfos extends HapiQueryOp<HapiGetAccountNftInfos> {
+    private static final Logger log = LogManager.getLogger(HapiGetAccountNftInfos.class);
+
+    private String account;
+    private long start;
+    private long end;
+
+    public HapiGetAccountNftInfos(String account, long start, long end) {
+        this.account = account;
+        this.start = start;
+        this.end = end;
+    }
+
+    private Optional<List<HapiTokenNftInfo>> expectedNfts = Optional.empty();
+
+    @Override
+    public HederaFunctionality type() {
+        return HederaFunctionality.TokenGetNftInfo;
+    }
+
+    @Override
+    protected HapiGetAccountNftInfos self() {
+        return this;
+    }
+
+    @Override
+    protected void assertExpectationsGiven(HapiApiSpec spec) throws Throwable {
```

Comment:
There is a bug in the assertion. We have to make sure that the whole list is equal to the expected one. Atm we are searching for an Nft in the expected list and that has problems if we mint 2 different Nft (`1` and `2) and try to assert that the minted nfts are `1` and `1`

Reply:
resolved with #1724